File: src/main/java/dev/paigewatson/layoutmaster/models/rollingstock/RollingStock.java
package dev.paigewatson.layoutmaster.models.rollingstock;

import dev.paigewatson.layoutmaster.models.NullableEntity;
import dev.paigewatson.layoutmaster.models.goods.GoodsType;

public interface RollingStock extends NullableEntity
{
    boolean isAARType(AARDesignation expectedAARDesignation);

    boolean isCarrying(GoodsType goodsType);

    void load(GoodsType goodsType);
}

File: src/main/java/dev/paigewatson/layoutmaster/models/rollingstock/RollingStock.kt
package dev.paigewatson.layoutmaster.models.rollingstock

import dev.paigewatson.layoutmaster.models.NullableEntity
import dev.paigewatson.layoutmaster.models.goods.GoodsType

interface RollingStock : NullableEntity {
    fun isAARType(expectedAARDesignation: AARDesignation): Boolean
    fun isCarrying(goodsType: GoodsType): Boolean
    fun load(goodsType: GoodsType)
}
